upgrade_analysis/tests/test_module.py

```diff
@@ -0,0 +1,46 @@
+from odoo.tests import common, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestUpgradeAnalysis(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.IrModuleModule = self.env["ir.module.module"]
+        self.product_module = self.IrModuleModule.search([("name", "=", "product")])
+        self.sale_module = self.IrModuleModule.search([("name", "=", "sale")])
+        self.upgrade_analysis = self.IrModuleModule.search(
+            [("name", "=", "upgrade_analysis")]
+        )
+
+    def test_upgrade_install_wizard(self):
+        InstallWizard = self.env["upgrade.install.wizard"]
+        wizard = InstallWizard.create({})
+
+        wizard.select_odoo_modules()
+        self.assertTrue(
+            self.product_module.id in wizard.module_ids.ids,
+            "Select Odoo module should select 'product' module",
+        )
+
+        wizard.select_oca_modules()
+        self.assertTrue(
+            self.upgrade_analysis.id in wizard.module_ids.ids,
+            "Select OCA module should select 'upgrade_analysis' module",
+        )
+
+        wizard.select_other_modules()
+        self.assertTrue(
+            self.product_module.id not in wizard.module_ids.ids,
+            "Select Other module should not select 'product' module",
+        )
```

Comment:
```suggestion
        self.assertFalse(
            self.product_module.id in wizard.module_ids.ids,
            "Select Other module should not select 'product' module",
        )
```

https://www.recallstack.icu/en/2020/09/01/do-not-use-negative-booleans-please/